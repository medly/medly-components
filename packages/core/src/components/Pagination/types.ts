import { HTMLProps } from '@medly-components/utils';

export interface PaginationProps extends HTMLProps<HTMLUListElement> {
    /** Active page number */
    activePage?: number;
    /** Total items to be paginated */
    totalItems: number;
    /** Function to be called on page click */
    onPageClick: (page: number) => void;
    /** Items count per page */
    itemsPerPage?: number;
    /** Hide Next and Prev named button */
    hidePrevNextLinks?: boolean;
    /** Page selector popup placement */
    pageSelectorPopupPlacement?: 'top' | 'bottom';
}
