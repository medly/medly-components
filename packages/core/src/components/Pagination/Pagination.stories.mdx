import { Pagination } from './Pagination';
import { defaultTheme } from '@medly-components/theme';
import { boolean, number } from '@storybook/addon-knobs';
import { Preview, Story, Meta, Props } from '@storybook/addon-docs/blocks';
import { ThemeInterface, DummyWrapper } from './Pagination.stories';
import { useState } from 'react';
import { Placement } from '../Popover/types';

<Meta title="Core" component={Pagination} />

# Pagination

The `Pagination` component allows you to show the data in a sequence of pages and select a specific page from a range of pages.

<Preview withToolbar>
    <Story name="Pagination">
        {() => {
            const [activePage, setActivePage] = useState(1);
            return (
                <DummyWrapper>
                    <Pagination
                        activePage={activePage}
                        totalItems={number('Total Items', 1000)}
                        itemsPerPage={number('Items per page', 20)}
                        hidePrevNextLinks={boolean('Hide Prev Next Links', false)}
                        onPageClick={setActivePage}
                    />
                </DummyWrapper>
            );
        }}
    </Story>
</Preview>

## Props

This component offers the following props mentioned in the table below:

<Props of={Pagination} />

## Theme

This component offers the following themes mentioned in the table below:

<Props of={ThemeInterface} />
