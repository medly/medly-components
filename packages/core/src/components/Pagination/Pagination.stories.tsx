import { defaultTheme, PaginationTheme } from '@medly-components/theme';
import { centerAligned } from '@medly-components/utils';
import styled from 'styled-components';
import type { FC } from 'react';

export const ThemeInterface: FC<PaginationTheme> = () => null;
ThemeInterface.defaultProps = {
    ...defaultTheme.pagination
};

export const DummyWrapper = styled('div')`
    width: 100%;
    height: 100%;
    ${centerAligned()};
`;
