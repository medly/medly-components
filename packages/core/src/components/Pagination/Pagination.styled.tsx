import { ChevronLeftIcon, ChevronRightIcon, SvgIcon } from '@medly-components/icons';
import { css, getFontStyle, styled } from '@medly-components/utils';
import List from '../List';
import Text from '../Text';

export const ListWrapper = styled(List)`
    & > li {
        margin: 0;
    }
`;

export const getPageNumberButtonStyleByState = (
    itemKey: 'pageNumber' | 'overlayPageNumber',
    state: 'default' | 'active' | 'pressed' | 'hovered'
) => css`
    color: ${({ theme: { pagination } }) => pagination[itemKey].color[state]};
    background: ${({ theme: { pagination } }) => pagination[itemKey].bgColor[state]};
`;

const getPageNavButtonStyleByState = (state: 'default' | 'pressed' | 'hovered' | 'disabled') => css`
    background: ${({ theme: { pagination } }) => pagination.pageNav.bgColor[state]};
    ${SvgIcon} {
        * {
            fill: ${({ theme: { pagination } }) => pagination.pageNav.color[state]};
        }
    }
`;

export const BaseButton = styled.button.attrs({ type: 'button' })`
    margin: 0.4rem;
    height: 3.2rem;
    width: 3.2rem;
    display: block;
    border-radius: 50%;
    user-select: none;
    border: none;
    padding: 0;
    text-align: center;
    transition: all 100ms ease-out;
    ${getPageNumberButtonStyleByState('pageNumber', 'default')}

    &:hover {
        cursor: pointer;
    }

    &:focus {
        outline: none;
    }

    &:disabled {
        cursor: not-allowed;
    }
`;
export const PageNumberButton = styled(BaseButton)<{ isActive?: boolean }>`
    ${props => getPageNumberButtonStyleByState('pageNumber', props.isActive ? 'active' : 'default')};
    border-radius: ${({ theme: { pagination } }) => pagination.pageNumber.borderRadius};

    &:hover {
        ${props => !props.isActive && getPageNumberButtonStyleByState('pageNumber', 'hovered')};
    }
    &:active {
        font-weight: ${({ theme }) => theme.font.weights.Strong};
        ${getPageNumberButtonStyleByState('pageNumber', 'pressed')};
    }
    ${Text.Style} {
        ${({ theme }) => getFontStyle({ theme, fontVariant: theme.pagination.fontVariant })};
        font-weight: ${({ theme, isActive }) =>
            theme.font.weights[theme.pagination.pageNumber.fontWeight[isActive ? 'selected' : 'default']]};
    }
`;

export const PageNavButton = styled(BaseButton)<{ disabled: boolean }>`
    & > ${ChevronLeftIcon.Style}, ${ChevronRightIcon.Style} {
        padding: 0.4rem;
    }

    ${getPageNavButtonStyleByState('default')}

    &:disabled {
        ${getPageNavButtonStyleByState('disabled')}
    }

    &:not(:disabled) {
        &:hover {
            ${getPageNavButtonStyleByState('hovered')}
        }
        &:active {
            ${getPageNavButtonStyleByState('pressed')}
        }
    }
`;
