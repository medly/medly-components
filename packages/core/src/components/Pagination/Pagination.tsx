import { ChevronLeftIcon, ChevronRightIcon } from '@medly-components/icons';
import { WithStyle } from '@medly-components/utils';
import React, { FC, useMemo } from 'react';
import Popover from '../Popover';
import Text from '../Text';
import { paginator } from './helper';
import { ListWrapper, PageNavButton, PageNumberButton } from './Pagination.styled';
import PaginationPopup from './PaginationPopup';
import { PaginationProps } from './types';

const Component: FC<PaginationProps> = React.memo(
    React.forwardRef((props, ref) => {
        const links = [],
            { hidePrevNextLinks, activePage, itemsPerPage, totalItems, onPageClick, ...restProps } = props,
            { currentPage, linkItems, totalPages } = useMemo(() => paginator(totalItems, activePage, itemsPerPage), [
                totalItems,
                activePage,
                itemsPerPage
            ]);

        const onClickHandler = (page: number | '...') => () => page !== '...' && page !== currentPage && onPageClick(page);

        for (let i = 0; i < linkItems.length; i++) {
            if (linkItems[i] === '...')
                links.push(
                    <Popover interactionType="click">
                        <PaginationPopup
                            prevPageNumber={linkItems[i - 1] as number}
                            nextPageNumber={linkItems[i + 1] as number}
                            onClickHandler={onPageClick}
                        ></PaginationPopup>
                    </Popover>
                );
            else
                links.push(
                    <PageNumberButton key={i} onClick={onClickHandler(linkItems[i])} isActive={linkItems[i] === currentPage}>
                        <Text textAlign="center">{linkItems[i]}</Text>
                    </PageNumberButton>
                );
        }

        if (!hidePrevNextLinks) {
            links.unshift(
                <PageNavButton key="first" disabled={currentPage < 2} onClick={onClickHandler(currentPage - 1)}>
                    <ChevronLeftIcon size="M" title="prev" />
                </PageNavButton>
            );
            links.push(
                <PageNavButton key="last" disabled={currentPage === totalPages} onClick={onClickHandler(currentPage + 1)}>
                    <ChevronRightIcon size="M" title="next" />
                </PageNavButton>
            );
        }

        return (
            <ListWrapper ref={ref} variant="horizontal" {...restProps}>
                {links}
            </ListWrapper>
        );
    })
);

Component.displayName = 'Pagination';
Component.defaultProps = { activePage: 1, itemsPerPage: 20, hidePrevNextLinks: false };
export const Pagination: FC<PaginationProps> & WithStyle = Object.assign(Component, { Style: ListWrapper.Style });
