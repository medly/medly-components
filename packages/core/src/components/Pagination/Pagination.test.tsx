import { cleanup, fireEvent, render } from '@test-utils';
import React from 'react';
import { Pagination } from './Pagination';

describe('Pagination component', () => {
    const renderer = (props: any = { totalItems: 0 }) => {
        const mockOnPageClick = jest.fn();
        return {
            mockOnPageClick,
            ...render(<Pagination {...props} onPageClick={mockOnPageClick} />)
        };
    };

    afterEach(cleanup);

    it('should render correctly with all the default props', () => {
        const { container } = renderer({ totalItems: 150 });
        expect(container).toMatchSnapshot();
    });

    it('should render correctly when total items is 0', () => {
        const { container } = renderer();
        expect(container).toMatchSnapshot();
    });

    it('should render correctly with all the props given', () => {
        const { container } = renderer({ totalItems: 150, activePage: 1, itemsPerPage: 10, hidePrevNextLinks: true });
        expect(container).toMatchSnapshot();
    });

    it('should call onClick handler with correct page number when any page link is clicked', () => {
        const { getByRole, mockOnPageClick } = renderer({ totalItems: 150, activePage: 3, itemsPerPage: 10 });
        fireEvent.click(getByRole('button', { name: '4' }));
        expect(mockOnPageClick).toBeCalledWith(4);
    });

    it('should call onClick handler with first page when first page is clicked', () => {
        const { getByRole, mockOnPageClick } = renderer({ totalItems: 150, activePage: 3, itemsPerPage: 10 });
        fireEvent.click(getByRole('button', { name: '1' }));
        expect(mockOnPageClick).toBeCalledWith(1);
    });

    it('should call onClick handler with last page when last page is clicked', () => {
        const { getByRole, mockOnPageClick } = renderer({ totalItems: 150, activePage: 3, itemsPerPage: 10 });
        fireEvent.click(getByRole('button', { name: '15' }));
        expect(mockOnPageClick).toBeCalledWith(15);
    });

    it('should call onClick handler with prev page when prev link is clicked', () => {
        const { mockOnPageClick, getByTitle } = renderer({ totalItems: 150, activePage: 3, itemsPerPage: 10 });
        fireEvent.click(getByTitle('prev'));
        expect(mockOnPageClick).toBeCalledWith(2);
    });

    it('should call onClick handler with next page when next link is clicked', () => {
        const { mockOnPageClick, getByTitle } = renderer({ totalItems: 150, activePage: 3, itemsPerPage: 10 });
        fireEvent.click(getByTitle('next'));
        expect(mockOnPageClick).toBeCalledWith(4);
    });

    it('should call onClick handler with expected page when selected from ellipsis popover', () => {
        const { mockOnPageClick, getByRole } = renderer({ totalItems: 150, activePage: 3, itemsPerPage: 10 });
        fireEvent.click(getByRole('button', { name: '...' }));
        fireEvent.click(getByRole('button', { name: '7' }));
        expect(mockOnPageClick).toBeCalledWith(7);
    });

    it('should not call onClick handler when clicked on active page', () => {
        const { mockOnPageClick, getByRole } = renderer({ totalItems: 150, activePage: 3, itemsPerPage: 10 });
        fireEvent.click(getByRole('button', { name: '3' }));
        expect(mockOnPageClick).not.toBeCalled();
    });
});
