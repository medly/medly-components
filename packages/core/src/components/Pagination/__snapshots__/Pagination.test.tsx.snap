// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination component should render correctly when total items is 0 1`] = `
.c4 {
  overflow: visible;
  font-size: 2.4rem;
  -webkit-transition: all 100ms linear;
  transition: all 100ms linear;
  cursor: inherit;
}

.c4 * {
  fill-opacity: 1;
  -webkit-transition: all 100ms linear;
  transition: all 100ms linear;
  fill: #607890;
}

.c0 {
  padding: 0;
  margin: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  list-style: none;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c0 li {
  margin: 0 0.4rem;
}

.c0 li:last-child {
  margin-right: 0.8rem;
}

.c0 li:first-child {
  margin-left: 0.8rem;
}

.c1 > li {
  margin: 0;
}

.c2 {
  margin: 0.4rem;
  height: 3.2rem;
  width: 3.2rem;
  display: block;
  border-radius: 50%;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  border: none;
  padding: 0;
  text-align: center;
  -webkit-transition: all 100ms ease-out;
  transition: all 100ms ease-out;
  color: #435465;
  background: #ffffff;
  background: #ffffff;
}

.c2:hover {
  cursor: pointer;
}

.c2:focus {
  outline: none;
}

.c2:disabled {
  cursor: not-allowed;
}

.c2 > .c3,
.c2 .c3 {
  padding: 0.4rem;
}

.c2 .c3 * {
  fill: #435465;
}

.c2:disabled {
  background: #ffffff;
}

.c2:disabled .c3 * {
  fill: #C7D0D8;
}

.c2:not(:disabled):hover {
  background: #F7F8F9;
}

.c2:not(:disabled):hover .c3 * {
  fill: #303C48;
}

.c2:not(:disabled):active {
  background: #eff2f4;
}

.c2:not(:disabled):active .c3 * {
  fill: #13181D;
}

<div>
  <ul
    class="c0 c1"
  >
    <li>
      <button
        class="sc-1dmgj8o-1 c2"
        disabled=""
        type="button"
      >
        <svg
          class="c3 c4"
          fill="none"
          height="1em"
          title="prev"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M15 6.705a.996.996 0 0 0-1.41 0L9 11.295a.996.996 0 0 0 0 1.41l4.59 4.59a.996.996 0 1 0 1.41-1.41l-3.88-3.89L15 8.115c.39-.39.38-1.03 0-1.41z"
            fill="#000"
            fill-opacity=".54"
          />
        </svg>
      </button>
    </li>
    <li>
      <button
        class="sc-1dmgj8o-1 c2"
        disabled=""
        type="button"
      >
        <svg
          class="c3 c4"
          fill="none"
          height="1em"
          title="next"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M9 6.71a.996.996 0 0 0 0 1.41L12.88 12 9 15.88a.996.996 0 1 0 1.41 1.41L15 12.7a.996.996 0 0 0 0-1.41L10.41 6.7c-.38-.38-1.02-.38-1.41.01z"
            fill="#000"
            fill-opacity=".54"
          />
        </svg>
      </button>
    </li>
  </ul>
</div>
`;

exports[`Pagination component should render correctly with all the default props 1`] = `
.c4 {
  overflow: visible;
  font-size: 2.4rem;
  -webkit-transition: all 100ms linear;
  transition: all 100ms linear;
  cursor: inherit;
}

.c4 * {
  fill-opacity: 1;
  -webkit-transition: all 100ms linear;
  transition: all 100ms linear;
  fill: #607890;
}

.c9 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.c7 {
  margin: 0;
  color: inherit;
  font-size: 1.4rem;
  font-weight: 400;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  line-height: 2.2rem;
  text-align: center;
}

.c10 {
  margin: 0;
  color: inherit;
  font-size: 1.2rem;
  font-weight: 600;
  -webkit-letter-spacing: 0.12rem;
  -moz-letter-spacing: 0.12rem;
  -ms-letter-spacing: 0.12rem;
  letter-spacing: 0.12rem;
  line-height: 2.0rem;
  text-align: center;
}

.c0 {
  padding: 0;
  margin: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  list-style: none;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c0 li {
  margin: 0 0.4rem;
}

.c0 li:last-child {
  margin-right: 0.8rem;
}

.c0 li:first-child {
  margin-left: 0.8rem;
}

.c1 > li {
  margin: 0;
}

.c5 {
  margin: 0.4rem;
  height: 3.2rem;
  width: 3.2rem;
  display: block;
  border-radius: 50%;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  border: none;
  padding: 0;
  text-align: center;
  -webkit-transition: all 100ms ease-out;
  transition: all 100ms ease-out;
  color: #435465;
  background: #ffffff;
  color: #3872D2;
  background: #EBF1FA;
  border-radius: 50%;
}

.c5:hover {
  cursor: pointer;
}

.c5:focus {
  outline: none;
}

.c5:disabled {
  cursor: not-allowed;
}

.c5:active {
  font-weight: 700;
  color: #13181D;
  background: #eff2f4;
}

.c5 .c6 {
  font-size: 1.2rem;
  font-weight: 700;
  -webkit-letter-spacing: 0.12rem;
  -moz-letter-spacing: 0.12rem;
  -ms-letter-spacing: 0.12rem;
  letter-spacing: 0.12rem;
  line-height: 2.0rem;
  font-family: Open Sans,Helvetica Neue,Helvetica,Arial,sans-serif;
  font-weight: 700;
}

.c8 {
  margin: 0.4rem;
  height: 3.2rem;
  width: 3.2rem;
  display: block;
  border-radius: 50%;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  border: none;
  padding: 0;
  text-align: center;
  -webkit-transition: all 100ms ease-out;
  transition: all 100ms ease-out;
  color: #435465;
  background: #ffffff;
  color: #435465;
  background: #ffffff;
  border-radius: 50%;
}

.c8:hover {
  cursor: pointer;
}

.c8:focus {
  outline: none;
}

.c8:disabled {
  cursor: not-allowed;
}

.c8:hover {
  color: #303C48;
  background: #F7F8F9;
}

.c8:active {
  font-weight: 700;
  color: #13181D;
  background: #eff2f4;
}

.c8 .c6 {
  font-size: 1.2rem;
  font-weight: 700;
  -webkit-letter-spacing: 0.12rem;
  -moz-letter-spacing: 0.12rem;
  -ms-letter-spacing: 0.12rem;
  letter-spacing: 0.12rem;
  line-height: 2.0rem;
  font-family: Open Sans,Helvetica Neue,Helvetica,Arial,sans-serif;
  font-weight: 600;
}

.c2 {
  margin: 0.4rem;
  height: 3.2rem;
  width: 3.2rem;
  display: block;
  border-radius: 50%;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  border: none;
  padding: 0;
  text-align: center;
  -webkit-transition: all 100ms ease-out;
  transition: all 100ms ease-out;
  color: #435465;
  background: #ffffff;
  background: #ffffff;
}

.c2:hover {
  cursor: pointer;
}

.c2:focus {
  outline: none;
}

.c2:disabled {
  cursor: not-allowed;
}

.c2 > .c3,
.c2 .c3 {
  padding: 0.4rem;
}

.c2 .c3 * {
  fill: #435465;
}

.c2:disabled {
  background: #ffffff;
}

.c2:disabled .c3 * {
  fill: #C7D0D8;
}

.c2:not(:disabled):hover {
  background: #F7F8F9;
}

.c2:not(:disabled):hover .c3 * {
  fill: #303C48;
}

.c2:not(:disabled):active {
  background: #eff2f4;
}

.c2:not(:disabled):active .c3 * {
  fill: #13181D;
}

<div>
  <ul
    class="c0 c1"
  >
    <li>
      <button
        class="sc-1dmgj8o-1 c2"
        disabled=""
        type="button"
      >
        <svg
          class="c3 c4"
          fill="none"
          height="1em"
          title="prev"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M15 6.705a.996.996 0 0 0-1.41 0L9 11.295a.996.996 0 0 0 0 1.41l4.59 4.59a.996.996 0 1 0 1.41-1.41l-3.88-3.89L15 8.115c.39-.39.38-1.03 0-1.41z"
            fill="#000"
            fill-opacity=".54"
          />
        </svg>
      </button>
    </li>
    <li>
      <button
        class="sc-1dmgj8o-1 c5"
        type="button"
      >
        <span
          class="c6 c7"
        >
          1
        </span>
      </button>
    </li>
    <li>
      <button
        class="sc-1dmgj8o-1 c8"
        type="button"
      >
        <span
          class="c6 c7"
        >
          2
        </span>
      </button>
    </li>
    <li>
      <button
        class="sc-1dmgj8o-1 c8"
        type="button"
      >
        <span
          class="c6 c7"
        >
          3
        </span>
      </button>
    </li>
    <li>
      <div
        class="c9"
        id="medly-popover-wrapper"
      >
        <button
          class="sc-1dmgj8o-1 c8"
          type="button"
        >
          <h5
            class="c6 c10"
          >
            ...
          </h5>
        </button>
      </div>
    </li>
    <li>
      <button
        class="sc-1dmgj8o-1 c8"
        type="button"
      >
        <span
          class="c6 c7"
        >
          8
        </span>
      </button>
    </li>
    <li>
      <button
        class="sc-1dmgj8o-1 c2"
        type="button"
      >
        <svg
          class="c3 c4"
          fill="none"
          height="1em"
          title="next"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M9 6.71a.996.996 0 0 0 0 1.41L12.88 12 9 15.88a.996.996 0 1 0 1.41 1.41L15 12.7a.996.996 0 0 0 0-1.41L10.41 6.7c-.38-.38-1.02-.38-1.41.01z"
            fill="#000"
            fill-opacity=".54"
          />
        </svg>
      </button>
    </li>
  </ul>
</div>
`;

exports[`Pagination component should render correctly with all the props given 1`] = `
.c6 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.c4 {
  margin: 0;
  color: inherit;
  font-size: 1.4rem;
  font-weight: 400;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  line-height: 2.2rem;
  text-align: center;
}

.c7 {
  margin: 0;
  color: inherit;
  font-size: 1.2rem;
  font-weight: 600;
  -webkit-letter-spacing: 0.12rem;
  -moz-letter-spacing: 0.12rem;
  -ms-letter-spacing: 0.12rem;
  letter-spacing: 0.12rem;
  line-height: 2.0rem;
  text-align: center;
}

.c0 {
  padding: 0;
  margin: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  list-style: none;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c0 li {
  margin: 0 0.4rem;
}

.c0 li:last-child {
  margin-right: 0.8rem;
}

.c0 li:first-child {
  margin-left: 0.8rem;
}

.c1 > li {
  margin: 0;
}

.c2 {
  margin: 0.4rem;
  height: 3.2rem;
  width: 3.2rem;
  display: block;
  border-radius: 50%;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  border: none;
  padding: 0;
  text-align: center;
  -webkit-transition: all 100ms ease-out;
  transition: all 100ms ease-out;
  color: #435465;
  background: #ffffff;
  color: #3872D2;
  background: #EBF1FA;
  border-radius: 50%;
}

.c2:hover {
  cursor: pointer;
}

.c2:focus {
  outline: none;
}

.c2:disabled {
  cursor: not-allowed;
}

.c2:active {
  font-weight: 700;
  color: #13181D;
  background: #eff2f4;
}

.c2 .c3 {
  font-size: 1.2rem;
  font-weight: 700;
  -webkit-letter-spacing: 0.12rem;
  -moz-letter-spacing: 0.12rem;
  -ms-letter-spacing: 0.12rem;
  letter-spacing: 0.12rem;
  line-height: 2.0rem;
  font-family: Open Sans,Helvetica Neue,Helvetica,Arial,sans-serif;
  font-weight: 700;
}

.c5 {
  margin: 0.4rem;
  height: 3.2rem;
  width: 3.2rem;
  display: block;
  border-radius: 50%;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  border: none;
  padding: 0;
  text-align: center;
  -webkit-transition: all 100ms ease-out;
  transition: all 100ms ease-out;
  color: #435465;
  background: #ffffff;
  color: #435465;
  background: #ffffff;
  border-radius: 50%;
}

.c5:hover {
  cursor: pointer;
}

.c5:focus {
  outline: none;
}

.c5:disabled {
  cursor: not-allowed;
}

.c5:hover {
  color: #303C48;
  background: #F7F8F9;
}

.c5:active {
  font-weight: 700;
  color: #13181D;
  background: #eff2f4;
}

.c5 .c3 {
  font-size: 1.2rem;
  font-weight: 700;
  -webkit-letter-spacing: 0.12rem;
  -moz-letter-spacing: 0.12rem;
  -ms-letter-spacing: 0.12rem;
  letter-spacing: 0.12rem;
  line-height: 2.0rem;
  font-family: Open Sans,Helvetica Neue,Helvetica,Arial,sans-serif;
  font-weight: 600;
}

<div>
  <ul
    class="c0 c1"
  >
    <li>
      <button
        class="sc-1dmgj8o-1 c2"
        type="button"
      >
        <span
          class="c3 c4"
        >
          1
        </span>
      </button>
    </li>
    <li>
      <button
        class="sc-1dmgj8o-1 c5"
        type="button"
      >
        <span
          class="c3 c4"
        >
          2
        </span>
      </button>
    </li>
    <li>
      <button
        class="sc-1dmgj8o-1 c5"
        type="button"
      >
        <span
          class="c3 c4"
        >
          3
        </span>
      </button>
    </li>
    <li>
      <div
        class="c6"
        id="medly-popover-wrapper"
      >
        <button
          class="sc-1dmgj8o-1 c5"
          type="button"
        >
          <h5
            class="c3 c7"
          >
            ...
          </h5>
        </button>
      </div>
    </li>
    <li>
      <button
        class="sc-1dmgj8o-1 c5"
        type="button"
      >
        <span
          class="c3 c4"
        >
          15
        </span>
      </button>
    </li>
  </ul>
</div>
`;
